packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFNavigationDelegateHostApi.m

```diff
@@ -5,17 +5,77 @@
 #import "FWFNavigationDelegateHostApi.h"
 #import "FWFWebViewConfigurationHostApi.h"
 
+@interface FWFNavigationDelegateFlutterApiImpl ()
+@property(nonatomic, weak) FWFInstanceManager *instanceManager;
+@end
+
+@implementation FWFNavigationDelegateFlutterApiImpl
+- (instancetype)initWithBinaryMessenger:(id<FlutterBinaryMessenger>)binaryMessenger
+                        instanceManager:(FWFInstanceManager *)instanceManager {
+  self = [self initWithBinaryMessenger:binaryMessenger];
+  if (self) {
+    _instanceManager = instanceManager;
+  }
+  return self;
+}
+
+- (void)didFinishNavigationFunction:(void (^)(WKWebView *, NSString *))function
+                            webView:(WKWebView *)webView
+                                URL:(NSString *)URL {
+  long functionIdentifier = [self.instanceManager identifierForInstance:function];
+  if (functionIdentifier != NSNotFound) {
+    [self didFinishNavigationFunctionWithIdentifier:@(functionIdentifier)
+                                  webViewIdentifier:@([self.instanceManager
+                                                        identifierForInstance:webView])
+                                                URL:URL
+                                         completion:^(NSError *error) {
+                                           if (error) {
+                                             NSLog(@"%@", error.description);
```

Comment:
Why is this logging instead of sending the error somewhere?

Reply:
I wasn't sure what else to do with the error. This occurs when the Dart code fails to handle the callback method or the `FlutterBinaryMessenger` fails to send the message. Typically, most plugins have just been eating these kind of errors, but I assumed it would be better to at least print it out. A possible alternative would be to have a general `FlutterApi` that waits for errors received by the host api? However, that could also return an error and then we are in the same situation.